### PR TITLE
Clean up the OSB client

### LINKED ISSUE #883: Deprovision sends a request body
Deprovision is a `DELETE` operation, which should not send a request body. Additionally, the [deprovisioning section of the OSB API spec](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#deprovisioning) specifies some query string parameters that we are not sending.

### LINKED ISSUE #884: Unbind is not sending query string parameters
According to [the OSB API spec on unbind](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#unbinding), `service_id` and `plan_id` must be sent in the query string. We are not sending those.

### PR DESCRIPTION
**Pull request by krancour**
Fixes #883 
Fixes #884 
Replaces #885 (which was reverted)
Replaces #886 (which doesn't work the way it appears)

* Stops sending request body on delete
* Deals with query parameters in a more idiomatic way
    * The previous way worked only via use of format strings and that does not at all account for query parameter values that must be url-encoded

### CODE REVIEW

File: pkg/brokerapi/openservicebroker/open_service_broker_client.go [hunk 1]
```diff
@@ -140,16 +135,18 @@ func (c *openServiceBrokerClient) GetCatalog() (*brokerapi.Catalog, error) {
...
-	var serviceInstanceURL string
-
-	if req.AcceptsIncomplete {
-		serviceInstanceURL = fmt.Sprintf(serviceInstanceAsyncFormatString, c.url, ID)
-	} else {
-		serviceInstanceURL = fmt.Sprintf(serviceInstanceFormatString, c.url, ID)
-	}
+	serviceInstanceURL := fmt.Sprintf(serviceInstanceFormatString, c.url, ID)
 
 	// TODO: Handle the auth
```
Comment (vaikas-google) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: I think this TODO is no longer necessary?

File: pkg/brokerapi/openservicebroker/open_service_broker_client.go [hunk 2]
```diff
@@ -185,16 +182,20 @@ func (c *openServiceBrokerClient) UpdateServiceInstance(ID string, req *brokerap
...
-	var serviceInstanceURL string
-
-	if req.AcceptsIncomplete {
-		serviceInstanceURL = fmt.Sprintf(serviceInstanceDeleteAsyncFormatString, c.url, ID, req.ServiceID, req.PlanID)
-	} else {
-		serviceInstanceURL = fmt.Sprintf(serviceInstanceDeleteFormatString, c.url, ID, req.ServiceID, req.PlanID)
-	}
+	serviceInstanceURL := fmt.Sprintf(serviceInstanceFormatString, c.url, ID)
 
 	// TODO: Handle the auth
```
Comment (vaikas-google) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: Same here, remove it?

File: pkg/brokerapi/openservicebroker/open_service_broker_client.go [hunk 3]
```diff
@@ -293,14 +307,25 @@ func (c *openServiceBrokerClient) DeleteServiceBinding(instanceID, bindingID, se
...
+	resp, err := sendOSBRequest(
+		c,
+		http.MethodGet,
+		url,
+		map[string]string{
+			"service_id": req.ServiceID,
+			"plan_id":    req.PlanID,
+			"operation":  req.Operation,
+		},
+		pollReq,
```
Comment (vaikas-google) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: We should not be sending a body here. nil here and remove L317.

**Comment by vaikas-google** [comment]
Also once you remove sending the body in last operation poll, you need to remove L102 in fake_** that fails if there's no body.

**Comment by pmorie** [comment]
I'm glad to see this patch, but I do not think that the broker api client should live in our repo long term, or for much longer. I've been whittling on a replacement for our client here: https://github.com/pmorie/go-open-service-broker-client (note: this is WIP and i have taken a couple days off. Should have test coverge at about 90% in the next couple days).  I will definitely be incorporating the changes in this PR.

**Comment by pmorie** [comment]
For the record, I do not mean to imply at all that i am against is patch, just trying to disclose fully :-)

File: pkg/brokerapi/openservicebroker/open_service_broker_client.go [hunk 4]
```diff
@@ -140,16 +135,17 @@ func (c *openServiceBrokerClient) GetCatalog() (*brokerapi.Catalog, error) {
...
-	// TODO: Handle the auth
-	resp, err := sendOSBRequest(c, http.MethodPut, serviceInstanceURL, req)
+	serviceInstanceURL := fmt.Sprintf(serviceInstanceFormatString, c.url, ID)
+
+	resp, err := sendOSBRequest(
+		c,
+		http.MethodPut,
+		serviceInstanceURL,
+		map[string]string{
+			"accepts_incomplete": fmt.Sprintf("%t", req.AcceptsIncomplete),
```
Comment (duglin) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: Would be kind of nice if we didn't send the query param at all in the 'false' case but I guess we could do that later.

**Comment by duglin** [comment]
LGTM if CI is happy

**Pull request merged**